include/libtorrent/torrent_info.hpp

```diff
@@ -146,30 +152,46 @@ namespace libtorrent {
 		// an error occurs. These overloads are not available when building
 		// without exception support.
 		//
-		// The ``flags`` argument is currently unused.
+		// The overload that takes a ``span`` also needs an extra parameter of
+		// type ``from_span_t`` to disambiguate the ``std::string`` overload for
+		// string literals. There is an object in the libtorrent namespace of this
+		// type called ``from_span``.
 #ifndef BOOST_NO_EXCEPTIONS
-		explicit torrent_info(bdecode_node const& torrent_file, int flags = 0);
-		explicit torrent_info(char const* buffer, int size, int flags = 0);
-		explicit torrent_info(std::string const& filename, int flags = 0);
+		explicit torrent_info(bdecode_node const& torrent_file);
+#ifndef TORRENT_NO_DEPRECATE
+		TORRENT_DEPRECATED
+		torrent_info(char const* buffer, int size, int = 0)
+			: torrent_info(span<char const>{buffer, std::size_t(size)}, from_span) {}
+#endif
+		explicit torrent_info(span<char const> buffer, from_span_t);
```

Comment:
yes, I agree. I think the string overload is quite commonly used. In the long run, I would like to disambiguate them by using ``span<std::byte const>`` instead of ``char``.Perhaps that's even feasible in the short term